kong/conf_loader/init.lua

```diff
@@ -626,6 +656,63 @@ local function parse_value(value, typ)
 end
 
 
+-- Check if module is dynamic
+local function check_dynamic_module(mod_name)
+  local configure_line = ngx.config.nginx_configure()
+  local mod_re = [[^.*--add-dynamic-module=(.+\/]] .. mod_name .. [[(\s|$)).*$]]
+  return ngx.re.match(configure_line, mod_re, "oi") ~= nil
+end
+
+
+-- Lookup dynamic module object
+-- this function will lookup for the `mod_name` dynamic module in the following
+-- paths:
+--  - /usr/local/kong/modules -- default path for modules in container images
+--  - <nginx binary path>/../modules
+-- @param[type=string] mod_name The module name to lookup, without file extension
+local function lookup_dynamic_module_so(mod_name, kong_conf)
+  log.debug("looking up dynamic module %s", mod_name)
+
+  local mod_file = fmt("/usr/local/kong/modules/%s.so", mod_name)
```

Comment:
There are several bits of gateway code that contain a hardcoded reference to `/usr/local/kong`, and these could in theory be united with a constant. However, things are a bit murky, because sometimes `/usr/local/kong` refers to the default prefix for runtime state, and other times it refers to the installation prefix.

So yes, this could be improved by using a constant, but doing so in a meaningful way that makes the code more sensible/understandable is not so trivial, because I think it really ought to involve re-thinking the ways we are using `/usr/local/kong` across the board. I'll opt to leave things as-is for this PR.

Reply:
I see. Up until a few years ago (before ngx_wasm_module), the only occurrence of `/usr/local/kong` was in the CLI because we hard-coded a few common installation paths for detecting the prefix. As with many other patterns that have degraded in this codebase, I suspect it was copy-pasted without understanding and propagated.